app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,16 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = current_user.comments.new(comment_params)
+    @post = Post.find(params[:post_id])
+    @comment.post_id = @post.id
+    @comment.save
+    flash[:success] = 'Commented!'
+    redirect_to "/users/#{current_user.id}/posts"
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
```

Comment:
- [OPTIONAL] The create method looks fine but optionally I would suggest you adding if else functions in case the comment cannot be saved, here is an example:
```
	def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end

```